### DISCUSSION TITLE
Limiting Coupons to one use per customer

### DISCUSSION THREAD

**Discussion by byron-roots** [duncanmcclean/simple-commerce#1107] [Help] [answered]
We seem to be having some issues with coupons. Customer wants to use a welcome discount which can only be used once, but when creating a coupon it appears that a user can use it as many times as the like, and the limit option only limits total uses irrespective of customer.

Are we setting this up wrong or is this the expected behaviour. If it is how can we limit a coupon to one use per user?

**Comment by duncanmcclean** [comment, marked as answer]
Currently, there's no way to limit a coupon to be used once per customer, sorry! It's something I'm planning on implementing in a future version of Simple Commerce though but I don't have an ETA on it.

For now, maybe we could add a [hook](https://statamic.dev/extending/hooks#about) into the [`Coupon::isValid()`](https://github.com/duncanmcclean/simple-commerce/blob/7.x/src/Coupons/Coupon.php#L96) method so you can add your own handling logic for coupons, which would allow you to workaround it for now.

**Reply by byron-roots** [reply]
Thanks Duncan, I'l have a look into adding a hook
